File: BW/10월3주차/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    // 남은 작업 일 수 저장
    for(i=0; i<progresses.length; i++){
        var value = speeds[i]
        var cnt = 1;
        while(100-progresses[i] > value){
            value += speeds[i];
            cnt++;
        }
        answer[i] = cnt;
    }
    // 남은 작업 일 수 비교
    cnt = 1;
    var result = [];
    var idx1 = 0;
    for(i=1; i<answer.length; i++){
        if(answer[idx1] >= answer[i]){
            cnt++;
        }
        else{
            result.push(cnt);
            idx1 = i;
            cnt=1;
        }
    }
    
    result.push(cnt);
    return result;
}

File: JY/10월 4주차/기능개발.py
import math
from collections import deque
def solution(progresses, speeds):
    answer = []
    due_date_list = deque()
    
    for idx, progess in enumerate(progresses):
        due_date_list.append(math.ceil((100-progess)/speeds[idx]))
    
    while due_date_list:
        date = due_date_list.popleft()
        cnt = 1
        while len(due_date_list) != 0 and date >= due_date_list[0] :
            due_date_list.popleft()
            cnt+=1
        answer.append(cnt)
        
    return answer


# 미리 계산한다면 speeds
# remain_date = [7, 70, 45] 
# due_date_list = [7, 3, 9] => ceil.3
# while due_date_lsit:
# curr = due_date_lsit.pop()
# while curr >= due_date_lsit[-1]:
#     pop()
#     cnt+=1
